arch/nios2/include/kernel_arch_data.h

```diff
@@ -53,7 +53,7 @@ struct _kernel_arch {
 
 typedef struct _kernel_arch _kernel_arch_t;
 
-extern char _interrupt_stack[CONFIG_ISR_STACK_SIZE];
+extern K_THREAD_STACK_DEFINE(_interrupt_stack, CONFIG_ISR_STACK_SIZE);
```

Comment:
Nitpickery: it's a little weird to have referencing occurrences of _interrupt_stack declared by a macro with "DEFINE" in its name. Do you think it's worth adding a new:

```
#define K_THREAD_STACK_DECLARE(sym, size) extern K_THREAD_STACK_DEFINE(sym, size)
```

to kernel.h, to use here and elsewhere? If not, no worries.